test/unit/routes/static.test.ts

```diff
@@ -0,0 +1,47 @@
+import { promises as fs } from "fs"
+import * as path from "path"
+import { tmpdir } from "../../utils/helpers"
+import * as httpserver from "../../utils/httpserver"
+import * as integration from "../../utils/integration"
+
+describe("static", () => {
+  let codeServer: httpserver.HttpServer | undefined
+  let testFile: string | undefined
+  let testFileContent: string | undefined
+
+  beforeAll(async () => {
+    const testDir = await tmpdir("static")
+    testFile = path.join(testDir, "test")
+    testFileContent = "static file contents"
+    await fs.writeFile(testFile, testFileContent)
+  })
+
+  afterEach(async () => {
+    if (codeServer) {
+      await codeServer.close()
+      codeServer = undefined
+    }
+  })
+
+  it("/static unauthed", async () => {
+    codeServer = await integration.setup(["--auth=none"], "")
+    let resp = await codeServer.fetch("/static")
+    expect(resp.status).toBe(404)
+
+    resp = await codeServer.fetch(`/static/-${testFile}`)
+    expect(resp.status).toBe(200)
+
+    const content = await resp.text()
+    expect(content).toStrictEqual(testFileContent)
+  })
```

Comment:
question: a bit confused here. I know the test is testing the `/static` route when unuathed but why do we expect a `404` when you fetch the route and then a `200` when fetching a file?

And why the `/-file` instead of `/file for `/static/-${testFile}?

Reply:
The static endpoint expects a commit (for cache purposes, in this case I'm just using `-`) and then the full path of the file to serve.

So anything other than that will result in a 404 (since no path means no file). We could also do `/static/-`, `/static/-/some/path/that/probably/doesnot/exist`, etc.

Reply:
Oh TIL. I didn't know how it worked so thanks for explaining that!

> We could also do...

Your call!